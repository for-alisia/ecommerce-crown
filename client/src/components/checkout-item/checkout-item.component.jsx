/** Libraries */
import React from 'react';
import { connect } from 'react-redux';

/** Redux elements */
import {
  clearItemFromCart,
  addItem,
  removeItem,
} from '../../redux/cart/cart.actions';

/** Styles */
import {
  CheckoutItemContainer,
  ImageContainer,
  CellContainer,
  QtyContainer,
  ArrowContainer,
  ValueCellContainer,
  RemoveCellContainer,
} from './checkout-item.styles';

export const CheckoutItem = ({
  item,
  clearItemFromCart,
  addItem,
  removeItem,
}) => {
  const { name, price, quantity, imageUrl } = item;
  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={imageUrl} alt={name} />
      </ImageContainer>
      <CellContainer>{name}</CellContainer>
      <QtyContainer>
        <ArrowContainer onClick={() => removeItem(item)}>
          &#10094;
        </ArrowContainer>
        <ValueCellContainer>{quantity}</ValueCellContainer>
        <ArrowContainer onClick={() => addItem(item)}>&#10095;</ArrowContainer>
      </QtyContainer>
      <CellContainer>{price}</CellContainer>
      <RemoveCellContainer onClick={() => clearItemFromCart(item)}>
        &#10005;
      </RemoveCellContainer>
    </CheckoutItemContainer>
  );
};

export default connect(null, { clearItemFromCart, addItem, removeItem })(
  CheckoutItem
);
