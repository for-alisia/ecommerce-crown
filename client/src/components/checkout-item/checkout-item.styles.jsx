import styled, { css } from 'styled-components';

const commonWidthForSpan = css`
  width: 23%;
`;
export const CheckoutItemContainer = styled.div`
  width: 100%;
  display: flex;
  min-height: 100px;
  border-bottom: 1px solid darkgrey;
  padding: 15px 0;
  font-size: 20px;
  align-items: center;
`;

export const ImageContainer = styled.div`
  width: 23%;
  padding-right: 15px;

  img {
    width: 100%;
    height: 100%;
  }
`;

export const CellContainer = styled.span`
  ${commonWidthForSpan}
`;

export const QtyContainer = styled.span`
  ${commonWidthForSpan}
  display: flex;
`;

export const ArrowContainer = styled.div`
  cursor: pointer;
`;

export const ValueCellContainer = styled.span`
  margin: 0 10px;
`;

export const RemoveCellContainer = styled.div`
  padding-left: 12px;
  cursor: pointer;
`;
