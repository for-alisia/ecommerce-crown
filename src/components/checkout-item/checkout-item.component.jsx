/** Libraries */
import React from 'react';
import { connect } from 'react-redux';

/** Redux elements */
import {
  clearItemFromCart,
  addItem,
  removeItem,
} from '../../redux/cart/cart.actions';

/** Styles */
import './checkout-item.styles.scss';

const CheckoutItem = ({ item, clearItemFromCart, addItem, removeItem }) => {
  const { name, price, quantity, imageUrl } = item;
  return (
    <div className='checkout-item'>
      <div className='image-container'>
        <img src={imageUrl} alt={name} />
      </div>
      <span className='name'>{name}</span>
      <span className='quantity'>
        <div className='arrow' onClick={() => removeItem(item)}>
          &#10094;
        </div>
        <span className='value'>{quantity}</span>
        <div className='arrow' onClick={() => addItem(item)}>
          &#10095;
        </div>
      </span>
      <span className='price'>{price}</span>
      <div className='remove-button' onClick={() => clearItemFromCart(item)}>
        &#10005;
      </div>
    </div>
  );
};

export default connect(null, { clearItemFromCart, addItem, removeItem })(
  CheckoutItem
);
