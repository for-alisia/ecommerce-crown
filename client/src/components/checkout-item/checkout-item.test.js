// @ts-nocheck
/** Libraries */
import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';

/** Components */
import { CheckoutItem } from './checkout-item.component';

let wrapper;

beforeEach(() => {
  const mockProps = {
    item: {
      name: 'Test',
      price: 50,
      quantity: 2,
      imageUrl: 'https://some.com',
    },
    clearItemFromCart: jest.fn(),
    addItem: jest.fn(),
    removeItem: jest.fn(),
  };

  wrapper = shallow(<CheckoutItem {...mockProps} />);
});
describe('Checkout item component', () => {
  it('expects to match a snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
