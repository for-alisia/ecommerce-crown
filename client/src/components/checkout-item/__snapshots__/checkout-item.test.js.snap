// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout item component expects to match a snapshot 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="Test"
      src="https://some.com"
    />
  </styled.div>
  <styled.span>
    Test
  </styled.span>
  <styled.span>
    <styled.div
      onClick={[Function]}
    >
      ❮
    </styled.div>
    <styled.span>
      2
    </styled.span>
    <styled.div
      onClick={[Function]}
    >
      ❯
    </styled.div>
  </styled.span>
  <styled.span>
    50
  </styled.span>
  <styled.div
    onClick={[Function]}
  >
    ✕
  </styled.div>
</styled.div>
`;
